src/main.rs

```diff
@@ -130,11 +109,25 @@ fn main() -> Result<(), failure::Error> {
             commands::whoami(&user);
         }
 
-        if let Some(matches) = matches.subcommand_matches("publish") {
-            let name = matches.value_of("name");
+        if matches.subcommand_matches("build").is_some() {
+            commands::build(&cache, &user.settings.project.project_type)?;
+        }
+
+        if let Some(matches) = matches.subcommand_matches("preview") {
+            let method = HTTPMethod::from_str(matches.value_of("method").unwrap_or("get"));
+
+            let body = match matches.value_of("body") {
+                Some(s) => Some(s.to_string()),
+                None => None,
+            };
```

Comment:
This is

```rust
let body = matches.value_of("body").map(|s| s.to_string());
```

Reply:
yeah i think i actually like this but friggin clippy thinks ur suggestion is a redundant closure and has no problem with what's there atm